gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/metadata/crud_service/MetadataCrudService.java

```diff
@@ -16,7 +16,9 @@
 package io.gravitee.apim.core.metadata.crud_service;
 
 import io.gravitee.apim.core.metadata.model.Metadata;
+import java.util.Optional;
 
 public interface MetadataCrudService {
     Metadata create(Metadata metadata);
+    Optional<Metadata> findById(String referenceId, Metadata.ReferenceType referenceType, String key);
```

Comment:
I was not comfortable with the name `findById` but I remembered that in the data the id is this triplet (referenceId, referenceType and key)

I'm wondering if we should create a `MetadataId` record with those 3 attriubtes to make it explicit 🤔